Storybook cannot run in Angular CLI project that has been migrated to Nx

### Current Behavior

After creating an Angular project and integrating Nx, I added a ui-library (also angular) and added storybook. However after running storybook a lot of error messages came up. All of those error messages have to do with functions being declared multiple times, or types not compatible.

Error: node_modules/@types/react-dom/node_modules/@types/react/index.d.ts:3251:13 - error TS2717: Subsequent property declarations must have the same type.  Property 'feFlood' must be of type 'SVGProps<SVGFEFloodElement>', but here has type 'SVGProps<SVGFEFloodElement>'

node_modules/@types/react/index.d.ts:3401:13
    3401             feFuncG: React.SVGProps<SVGFEFuncGElement>;
                     ~~~~~~~
    'feFuncG' was also declared here.
    
etc etc.    

### Expected Behavior

I would expect storybook to run and open in the browser without errors.

### GitHub Repo

_No response_

### Steps to Reproduce

Using Nvm 18.16.1

1. npm init @angular abc
2. cd abc
3. npx nx@latest init
4. npm install -D @nx/angular
5. nx g @nx/angular:lib abc-ui
6. npm install -D @nx/storybook
7. nx g @nx/storybook:configuration abc-ui
8. nx run abc-ui:storybook


### Nx Report

```shell
Node   : 18.16.1
   OS     : darwin-x64
   npm    : 9.5.1

   nx (global)        : 16.5.2
   nx                 : 16.5.2
   @nx/js             : 16.5.2
   @nx/jest           : 16.5.2
   @nx/linter         : 16.5.2
   @nx/workspace      : 16.5.2
   @nx/angular        : 16.5.2
   @nx/cypress        : 16.5.2
   @nx/devkit         : 16.5.2
   @nx/eslint-plugin  : 16.5.2
   @nx/storybook      : 16.5.2
   @nrwl/tao          : 16.5.2
   @nx/webpack        : 16.5.2
   typescript         : 5.1.6
   ---------------------------------------
   Community plugins:
   @storybook/angular : 7.1.0
```


### Failure Logs

```shell
node_modules/@types/react/index.d.ts:3403:13
    3403             feGaussianBlur: React.SVGProps<SVGFEGaussianBlurElement>;
                     ~~~~~~~~~~~~~~
    'feGaussianBlur' was also declared here.

    at addError (/Users/project/abc/node_modules/@ngtools/webpack/src/ivy/diagnostics.js:31:29)
    at /Users/project/abc/node_modules/@ngtools/webpack/src/ivy/diagnostics.js:17:17
    at AngularWebpackPlugin.updateJitProgram (/Users/project/abc/node_modules/@ngtools/webpack/src/ivy/plugin.js:449:9)
    at AngularWebpackPlugin.setupCompilation (/Users/project/abc/node_modules/@ngtools/webpack/src/ivy/plugin.js:172:20)
    at /Users/project/abc/node_modules/@ngtools/webpack/src/ivy/plugin.js:103:22
    at Hook.eval [as call] (eval at create (/Users/project/abc/node_modules/tapable/lib/HookCodeFactory.js:19:10), <anonymous>:19:1)
    at Hook.CALL_DELEGATE [as _call] (/Users/project/abc/node_modules/tapable/lib/Hook.js:14:14)
    at Compiler.newCompilation (/Users/project/abc/node_modules/webpack/lib/Compiler.js:1123:30)
    at /Users/project/abc/node_modules/webpack/lib/Compiler.js:1168:29
    at _next0 (eval at create (/Users/project/abc/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:41:1)
Error: node_modules/@types/react-dom/node_modules/@types/react/index.d.ts:3257:13 - error TS2717: Subsequent property declarations must have the same type.  Property 'feImage' must be of type 'SVGProps<SVGFEImageElement>', but here has type 'SVGProps<SVGFEImageElement>'.

3257             feImage: React.SVGProps<SVGFEImageElement>;
                 ~~~~~~~

  node_modules/@types/react/index.d.ts:3404:13
    3404             feImage: React.SVGProps<SVGFEImageElement>;
                     ~~~~~~~
    'feImage' was also declared here.

    at addError (/Users/project/abc/node_modules/@ngtools/webpack/src/ivy/diagnostics.js:31:29)
    at /Users/project/abc/node_modules/@ngtools/webpack/src/ivy/diagnostics.js:17:17
    at AngularWebpackPlugin.updateJitProgram (/Users/project/abc/node_modules/@ngtools/webpack/src/ivy/plugin.js:449:9)
    at AngularWebpackPlugin.setupCompilation (/Users/project/abc/node_modules/@ngtools/webpack/src/ivy/plugin.js:172:20)
    at /Users/project/abc/node_modules/@ngtools/webpack/src/ivy/plugin.js:103:22
    at Hook.eval [as call] (eval at create (/Users/project/abc/node_modules/tapable/lib/HookCodeFactory.js:19:10), <anonymous>:19:1)
    at Hook.CALL_DELEGATE [as _call] (/Users/project/abc/node_modules/tapable/lib/Hook.js:14:14)
    at Compiler.newCompilation (/Users/project/abc/node_modules/webpack/lib/Compiler.js:1123:30)
    at /Users/project/abc/node_modules/webpack/lib/Compiler.js:1168:29
    at _next0 (eval at create (/Users/project/abc/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:41:1)
Error: node_modules/@types/react-dom/node_modules/@types/react/index.d.ts:3258:13 - error TS2717: Subsequent property declarations must have the same type.  Property 'feMerge' must be of type 'SVGProps<SVGFEMergeElement>', but here has type 'SVGProps<SVGFEMergeElement>'.

3258             feMerge: React.SVGProps<SVGFEMergeElement>;
                 ~~~~~~~

  node_modules/@types/react/index.d.ts:3405:13
    3405             feMerge: React.SVGProps<SVGFEMergeElement>;
                     ~~~~~~~
    'feMerge' was also declared here.

    at addError (/Users/project/abc/node_modules/@ngtools/webpack/src/ivy/diagnostics.js:31:29)
    at /Users/project/abc/node_modules/@ngtools/webpack/src/ivy/diagnostics.js:17:17
    at AngularWebpackPlugin.updateJitProgram (/Users/project/abc/node_modules/@ngtools/webpack/src/ivy/plugin.js:449:9)
    at AngularWebpackPlugin.setupCompilation (/Users/project/abc/node_modules/@ngtools/webpack/src/ivy/plugin.js:172:20)
    at /Users/project/abc/node_modules/@ngtools/webpack/src/ivy/plugin.js:103:22
    at Hook.eval [as call] (eval at create (/Users/project/abc/node_modules/tapable/lib/HookCodeFactory.js:19:10), <anonymous>:19:1)
    at Hook.CALL_DELEGATE [as _call] (/Users/project/abc/node_modules/tapable/lib/Hook.js:14:14)
    at Compiler.newCompilation (/Users/project/abc/node_modules/webpack/lib/Compiler.js:1123:30)
    at /Users/project/abc/node_modules/webpack/lib/Compiler.js:1168:29
    at _next0 (eval at create (/Users/project/abc/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:41:1)
Error: node_modules/@types/react-dom/node_modules/@types/react/index.d.ts:3259:13 - error TS2717: Subsequent property declarations must have the same type.  Property 'feMergeNode' must be of type 'SVGProps<SVGFEMergeNodeElement>', but here has type 'SVGProps<SVGFEMergeNodeElement>'.

3259             feMergeNode: React.SVGProps<SVGFEMergeNodeElement>;
                 ~~~~~~~~~~~

  node_modules/@types/react/index.d.ts:3406:13
    3406             feMergeNode: React.SVGProps<SVGFEMergeNodeElement>;
                     ~~~~~~~~~~~
    'feMergeNode' was also declared here.

    at addError (/Users/project/abc/node_modules/@ngtools/webpack/src/ivy/diagnostics.js:31:29)
    at /Users/project/abc/node_modules/@ngtools/webpack/src/ivy/diagnostics.js:17:17
    at AngularWebpackPlugin.updateJitProgram (/Users/project/abc/node_modules/@ngtools/webpack/src/ivy/plugin.js:449:9)
    at AngularWebpackPlugin.setupCompilation (/Users/project/abc/node_modules/@ngtools/webpack/src/ivy/plugin.js:172:20)
    at /Users/project/abc/node_modules/@ngtools/webpack/src/ivy/plugin.js:103:22
    at Hook.eval [as call] (eval at create (/Users/project/abc/node_modules/tapable/lib/HookCodeFactory.js:19:10), <anonymous>:19:1)
    at Hook.CALL_DELEGATE [as _call] (/Users/project/abc/node_modules/tapable/lib/Hook.js:14:14)
    at Compiler.newCompilation (/Users/project/abc/node_modules/webpack/lib/Compiler.js:1123:30)
    at /Users/project/abc/node_modules/webpack/lib/Compiler.js:1168:29
    at _next0 (eval at create (/Users/project/abc/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:41:1)
Error: node_modules/@types/react-dom/node_modules/@types/react/index.d.ts:3260:13 - error TS2717: Subsequent property declarations must have the same type.  Property 'feMorphology' must be of type 'SVGProps<SVGFEMorphologyElement>', but here has type 'SVGProps<SVGFEMorphologyElement>'.

3260             feMorphology: React.SVGProps<SVGFEMorphologyElement>;
                 ~~~~~~~~~~~~

  node_modules/@types/react/index.d.ts:3407:13
    3407             feMorphology: React.SVGProps<SVGFEMorphologyElement>;
                     ~~~~~~~~~~~~
    'feMorphology' was also declared here.

    at addError (/Users/project/abc/node_modules/@ngtools/webpack/src/ivy/diagnostics.js:31:29)
    at /Users/project/abc/node_modules/@ngtools/webpack/src/ivy/diagnostics.js:17:17
    at AngularWebpackPlugin.updateJitProgram (/Users/project/abc/node_modules/@ngtools/webpack/src/ivy/plugin.js:449:9)
    at AngularWebpackPlugin.setupCompilation (/Users/project/abc/node_modules/@ngtools/webpack/src/ivy/plugin.js:172:20)
    at /Users/project/abc/node_modules/@ngtools/webpack/src/ivy/plugin.js:103:22
    at Hook.eval [as call] (eval at create (/Users/project/abc/node_modules/tapable/lib/HookCodeFactory.js:19:10), <anonymous>:19:1)
    at Hook.CALL_DELEGATE [as _call] (/Users/project/abc/node_modules/tapable/lib/Hook.js:14:14)
    at Compiler.newCompilation (/Users/project/abc/node_modules/webpack/lib/Compiler.js:1123:30)
    at /Users/project/abc/node_modules/webpack/lib/Compiler.js:1168:29
    at _next0 (eval at create (/Users/project/abc/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:41:1)
Error: node_modules/@types/react-dom/node_modules/@types/react/index.d.ts:3261:13 - error TS2717: Subsequent property declarations must have the same type.  Property 'feOffset' must be of type 'SVGProps<SVGFEOffsetElement>', but here has type 'SVGProps<SVGFEOffsetElement>'.

3261             feOffset: React.SVGProps<SVGFEOffsetElement>;
                 ~~~~~~~~

  node_modules/@types/react/index.d.ts:3408:13
    3408             feOffset: React.SVGProps<SVGFEOffsetElement>;
                     ~~~~~~~~
    'feOffset' was also declared here.

    at addError (/Users/project/abc/node_modules/@ngtools/webpack/src/ivy/diagnostics.js:31:29)
    at /Users/project/abc/node_modules/@ngtools/webpack/src/ivy/diagnostics.js:17:17
    at AngularWebpackPlugin.updateJitProgram (/Users/project/abc/node_modules/@ngtools/webpack/src/ivy/plugin.js:449:9)
    at AngularWebpackPlugin.setupCompilation (/Users/project/abc/node_modules/@ngtools/webpack/src/ivy/plugin.js:172:20)
    at /Users/project/abc/node_modules/@ngtools/webpack/src/ivy/plugin.js:103:22
    at Hook.eval [as call] (eval at create (/Users/project/abc/node_modules/tapable/lib/HookCodeFactory.js:19:10), <anonymous>:19:1)
    at Hook.CALL_DELEGATE [as _call] (/Users/project/abc/node_modules/tapable/lib/Hook.js:14:14)
    at Compiler.newCompilation (/Users/project/abc/node_modules/webpack/lib/Compiler.js:1123:30)
    at /Users/project/abc/node_modules/webpack/lib/Compiler.js:1168:29
    at _next0 (eval at create (/Users/project/abc/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:41:1)
Error: node_modules/@types/react-dom/node_modules/@types/react/index.d.ts:3262:13 - error TS2717: Subsequent property declarations must have the same type.  Property 'fePointLight' must be of type 'SVGProps<SVGFEPointLightElement>', but here has type 'SVGProps<SVGFEPointLightElement>'.

3262             fePointLight: React.SVGProps<SVGFEPointLightElement>;
                 ~~~~~~~~~~~~

  node_modules/@types/react/index.d.ts:3409:13
    3409             fePointLight: React.SVGProps<SVGFEPointLightElement>;
                     ~~~~~~~~~~~~
    'fePointLight' was also declared here.

    at addError (/Users/project/abc/node_modules/@ngtools/webpack/src/ivy/diagnostics.js:31:29)
    at /Users/project/abc/node_modules/@ngtools/webpack/src/ivy/diagnostics.js:17:17
    at AngularWebpackPlugin.updateJitProgram (/Users/project/abc/node_modules/@ngtools/webpack/src/ivy/plugin.js:449:9)
    at AngularWebpackPlugin.setupCompilation (/Users/project/abc/node_modules/@ngtools/webpack/src/ivy/plugin.js:172:20)
    at /Users/project/abc/node_modules/@ngtools/webpack/src/ivy/plugin.js:103:22
    at Hook.eval [as call] (eval at create (/Users/project/abc/node_modules/tapable/lib/HookCodeFactory.js:19:10), <anonymous>:19:1)
    at Hook.CALL_DELEGATE [as _call] (/Users/project/abc/node_modules/tapable/lib/Hook.js:14:14)
    at Compiler.newCompilation (/Users/project/abc/node_modules/webpack/lib/Compiler.js:1123:30)
    at /Users/project/abc/node_modules/webpack/lib/Compiler.js:1168:29
    at _next0 (eval at create (/Users/project/abc/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:41:1)
Error: node_modules/@types/react-dom/node_modules/@types/react/index.d.ts:3263:13 - error TS2717: Subsequent property declarations must have the same type.  Property 'feSpecularLighting' must be of type 'SVGProps<SVGFESpecularLightingElement>', but here has type 'SVGProps<SVGFESpecularLightingElement>'.

3263             feSpecularLighting: React.SVGProps<SVGFESpecularLightingElement>;
                 ~~~~~~~~~~~~~~~~~~

  node_modules/@types/react/index.d.ts:3410:13
    3410             feSpecularLighting: React.SVGProps<SVGFESpecularLightingElement>;
                     ~~~~~~~~~~~~~~~~~~
    'feSpecularLighting' was also declared here.

    at addError (/Users/project/abc/node_modules/@ngtools/webpack/src/ivy/diagnostics.js:31:29)
    at /Users/project/abc/node_modules/@ngtools/webpack/src/ivy/diagnostics.js:17:17
    at AngularWebpackPlugin.updateJitProgram (/Users/project/abc/node_modules/@ngtools/webpack/src/ivy/plugin.js:449:9)
    at AngularWebpackPlugin.setupCompilation (/Users/project/abc/node_modules/@ngtools/webpack/src/ivy/plugin.js:172:20)
    at /Users/project/abc/node_modules/@ngtools/webpack/src/ivy/plugin.js:103:22
    at Hook.eval [as call] (eval at create (/Users/project/abc/node_modules/tapable/lib/HookCodeFactory.js:19:10), <anonymous>:19:1)
    at Hook.CALL_DELEGATE [as _call] (/Users/project/abc/node_modules/tapable/lib/Hook.js:14:14)
    at Compiler.newCompilation (/Users/project/abc/node_modules/webpack/lib/Compiler.js:1123:30)
    at /Users/project/abc/node_modules/webpack/lib/Compiler.js:1168:29
    at _next0 (eval at create (/Users/project/abc/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:41:1)
Error: node_modules/@types/react-dom/node_modules/@types/react/index.d.ts:3264:13 - error TS2717: Subsequent property declarations must have the same type.  Property 'feSpotLight' must be of type 'SVGProps<SVGFESpotLightElement>', but here has type 'SVGProps<SVGFESpotLightElement>'.
```


### Operating System

- [X] macOS
- [ ] Linux
- [ ] Windows
- [ ] Other (Please specify)

### Additional Information

Angular: 16.1.4

--- a/packages/storybook/src/generators/configuration/configuration.ts
+++ b/packages/storybook/src/generators/configuration/configuration.ts
@@ -23,6 +23,7 @@ import {
   configureTsSolutionConfig,
   createProjectStorybookDir,
   createStorybookTsconfigFile,
+  editTsconfigBaseJson,
   getE2EProjectName,
   getViteConfigFilePath,
   projectIsRootProjectInStandaloneWorkspace,
@@ -128,6 +129,7 @@ export async function configurationGenerator(
     );
   }
   configureTsProjectConfig(tree, schema);
+  editTsconfigBaseJson(tree);
   configureTsSolutionConfig(tree, schema);
   updateLintConfig(tree, schema);
 

--- a/packages/storybook/src/generators/configuration/lib/util-functions.ts
+++ b/packages/storybook/src/generators/configuration/lib/util-functions.ts
@@ -28,6 +28,7 @@ import {
 import { StorybookConfigureSchema } from '../schema';
 import { UiFramework7 } from '../../../utils/models';
 import { nxVersion } from '../../../utils/versions';
+import ts = require('typescript');
 
 const DEFAULT_PORT = 4400;
 
@@ -257,6 +258,23 @@ export function createStorybookTsconfigFile(
   };
 
   writeJson(tree, storybookTsConfigPath, storybookTsConfig);
+}
+
+export function editTsconfigBaseJson(tree: Tree) {
+  let tsconfigBasePath = 'tsconfig.base.json';
+
+  // standalone workspace maybe
+  if (!tree.exists(tsconfigBasePath)) tsconfigBasePath = 'tsconfig.json';
+
+  if (!tree.exists(tsconfigBasePath)) return;
+
+  const tsconfigBaseContent = readJson<TsConfig>(tree, tsconfigBasePath);
+
+  if (!tsconfigBaseContent.compilerOptions)
+    tsconfigBaseContent.compilerOptions = {};
+  tsconfigBaseContent.compilerOptions.skipLibCheck = true;
+
+  writeJson(tree, tsconfigBasePath, tsconfigBaseContent);
 }
 
 export function configureTsProjectConfig(

